Create AtCoderTools Intellij plugin

yuuki3655 opened the pull request:
This patch creates a new IntelliJ plugin for atcoder-tools integration.
The plugin is compatible with CLion and IntelliJ and it supports C++ and Java languages.

kyuridenamida commented:
Sorry for leaving this review for these two weeks.
LGTM although I'm not familiar with IntelliJ plugin development because the plugin is kind of isolated from the main repository. 
However, allow me to have one discussion before merging this.

One concern is the future support of this plugin, considering we don't have any integration test for this plugin and I'm not familiar with plugin development, it will be hard to maintain this tool without you.  I believe integration tests somewhat mitigate the problem.  I would like to ask your opinion on how difficult making integration tests for this plugin is & how much making them makes sense.

kyuridenamida commented on intellij-plugin/README.md:
```diff
@@ -0,0 +1,16 @@
+AtCoder Tools IntelliJ Plugin
+====
+IntelliJ Platformで動作するIDEからAtCoder Toolsを使用するためのPluginです。
```
これは非本質かもしれないんですがプラグインの導入に関する記述が少しあるとみんな使ってくれそうです。(バージョンによって違うと思うのであんまり事細かく説明したくないかもしれませんがインパクトはある気がします。

ところでこのプラグインをMarketplaceに出現させるのってどれくらい現実的なんですか

yuuki3655 replied:
おそらくアカウント作って、簡単な手続きをするだけだと思われます。https://plugins.jetbrains.com/docs/marketplace/

yuuki3655 commented:
> @yuuki3655
> Sorry for leaving this review for these two weeks.
> LGTM although I'm not familiar with IntelliJ plugin development because the plugin is kind of isolated from the main source codes.
> However, allow me to have one discussion before merging this.
> 
> One concern is the future support of this plugin, considering we don't have any integration test for this plugin and I'm not familiar with plugin development, it will be hard to maintain this tool without you. I believe integration tests somewhat mitigate this problem. I would like to ask your opinion on how difficult making integration tests for this plugin is & how much making them makes sense.

Sorry for leaving it for a long time. It's completely valid concern and I definitely need to write tests for this new component. Integration tests sound good approach. I think we need a fake implementation of atcoder tools to avoid sending real requests to the server. We can also add unit tests.

yuuki3655 commented:
Added some integration tests for core workflow for java and updated travis.yaml. Supporting for code coverage needs more work (I tried a bit but it seems we need some special handling for GuiTest). CLion plugin doesn't have integration test due to insufficient testing framework support. However, large portion of code are shared between CLion plugin and IntelliJ IDEA plugin and those are tested so I didn't worry about it too much now.

kyuridenamida commented:
これいつかマージしたくて、reopenしてもいいですか...

kyuridenamida commented:
@yuuki3655 ごめんなさい!結局(主にモチベが理由で)諦めました　5年放置してすみませんでした
